src/highdicom/sr/templates.py

```diff
@@ -489,6 +494,169 @@ def _contains_image_items(
+        '1.2.840.10008.5.1.4.1.1.88.33': codes.cid32.StructuredReportDocument,
+        '1.2.840.10008.5.1.4.1.1.88.34': codes.cid32.StructuredReportDocument,
+        '1.2.840.10008.5.1.4.1.1.88.35': codes.cid32.StructuredReportDocument,
+        '1.2.840.10008.5.1.4.1.1.88.50': codes.cid32.StructuredReportDocument,
+        '1.2.840.10008.5.1.4.1.1.88.59': codes.cid32.KeyObjectSelection,
+        '1.2.840.10008.5.1.4.1.1.88.65': codes.cid32.StructuredReportDocument,
+        '1.2.840.10008.5.1.4.1.1.88.67': codes.cid32.StructuredReportDocument,
+        '1.2.840.10008.5.1.4.1.1.88.68': codes.cid32.StructuredReportDocument,
+        '1.2.840.10008.5.1.4.1.1.88.70': codes.cid32.StructuredReportDocument,
+        '1.2.840.10008.5.1.4.1.1.88.71': codes.cid32.StructuredReportDocument,
+        '1.2.840.10008.5.1.4.1.1.88.72': codes.cid32.StructuredReportDocument,
+        '1.2.840.10008.5.1.4.1.1.88.73': codes.cid32.StructuredReportDocument,
+        '1.2.840.10008.5.1.4.1.1.88.74': codes.cid32.StructuredReportDocument,
+        '1.2.840.10008.5.1.4.1.1.88.75': codes.cid32.StructuredReportDocument,
+        '1.2.840.10008.5.1.4.1.1.88.76': codes.cid32.StructuredReportDocument,
+        '1.2.840.10008.5.1.4.1.1.90.1': codes.cid32.ContentAssessmentResult,
+        '1.2.840.10008.5.1.4.1.1.128': codes.cid29.PositronEmissionTomography,
+        '1.2.840.10008.5.1.4.1.1.130': codes.cid29.PositronEmissionTomography,
+        '1.2.840.10008.5.1.4.1.1.128.1': codes.cid29.PositronEmissionTomography,
+        '1.2.840.10008.5.1.4.1.1.200.2': codes.cid32.CTProtocol,
+        '1.2.840.10008.5.1.4.1.1.481.1': codes.cid29.RTImage,
+        '1.2.840.10008.5.1.4.1.1.481.2': codes.cid32.RTDose,
+        '1.2.840.10008.5.1.4.1.1.481.3': codes.cid32.RTStructureSet,
+        '1.2.840.10008.5.1.4.1.1.481.4': codes.cid32.RTTreatmentRecord,
+        '1.2.840.10008.5.1.4.1.1.481.5': codes.cid32.RTPlan,
+        '1.2.840.10008.5.1.4.1.1.481.6': codes.cid32.RTTreatmentRecord,
+        '1.2.840.10008.5.1.4.1.1.481.7': codes.cid32.RTTreatmentRecord,
+        '1.2.840.10008.5.1.4.1.1.481.8': codes.cid32.RTPlan,
+        '1.2.840.10008.5.1.4.1.1.481.9': codes.cid32.RTTreatmentRecord,
+    }
+    try:
+        return sopclass_to_modality_map[sop_class_uid]
+    except KeyError:
+        raise ValueError(
+            'SOP Class UID does not identify a SOP Class '
+            'for storage of an image information entity.'
+        )
+
+
+def _is_image(dataset: Dataset) -> bool:
+    """Check whether data set represents an image.
+
+    Parameters
+    ----------
+    dataset: pydicom.dataset.Dataset
+        Dataset
+
+    Returns
+    -------
+    bool
+        ``True`` if `dataset` is an image, ``False`` otherwise
+
+    """
+    if all(key in dataset for key in (
+        'Rows',
+        'Columns',
+        'SamplesPerPixel',
+        'PhotometricInterpretation',
+        'BitsAllocated'
+    )):
+        return True
+    else:
+        return False
```

Comment:
Note that there is now a function `highdicom._module_utils.does_iod_have_pixel_data` whose intent I think is very similar to this (determining whether a dataset represents an image when the pixel data may not actually be in the dataset). Could we use this here instead to reduce redundancy?

Reply:
Done. There was a similar request from Markus in the past but it lead to a recursive in Python.

Reply:
@CPBridge, @hackermd  - I've made this change but [note](https://github.com/herrmannlab/highdicom/pull/83#discussion_r656516824) this comment from Markus. He thought that a user might pass the Dataset without the PixelData for performance reasons. This was the reason that we were testing for non-PixelData members.

Reply:
Agreed that that is a consideration. When I wrote the `does_iod_have_pixel_data` function this was a goal too. That function doesn't check whether the *dataset* contains pixel data elements, it instead checks whether the *IOD* contains pixel data so it doesn't matter whether the pixel data has been dropped for performance reasons.